salt/modules/iptables.py

```diff
@@ -991,7 +991,7 @@ def delete(table, chain=None, position=None, rule=None, family="ipv4"):
     cmd = "{0} {1} -t {2} -D {3} {4}".format(
         _iptables_cmd(family), wait, table, chain, rule
     )
-    out = __salt__["cmd.run"](cmd)
+    out = __salt__["cmd.run_stderr"](cmd)
```

Comment:
Can you elaborate why we would not want stdout or stderr on some of these commands? If someone is using this execution module directly and not using the state module, they could be relying on getting information from stdout/stderr on some of these commands.

Reply:
@Ch3LL 
> Can you elaborate why we would not want stdout or stderr on some of these commands?

Sure, there are two reasons:

**In general**, because `cmd.run` intersperses `stdout` and `stderr` which will almost never be useful/safe for any type of parsing. In case such code exists, it should be rewritten using `cmd.run_*`. I am 99% convinced that any code that uses the return value of `cmd.run` in the codebase should be treated as a bug and `cmd.run` overall should be deprecated and long-term removed.

**In this case specifically**, because `iptables` returns an empty `stdout` on all of these calls _today_. Errors will be written to `stderr` via the `xtables_error -> basic_exit_err` macro (see https://git.netfilter.org/iptables/). 

Callers of `iptables.delete|flush|insert|append` almost certainly use this as "if the returned value is a non-empty string, you _might_ have encountered an error'. That again is terrible design, because you might also just have received an informational message from a future version, but being explicit about `stderr` being mostly errors today and `stdout` always being expected to be empty is an improvement.

For future reference: the most stable interface here would be returning the error code as that is an [enumerated list of error codes](https://git.netfilter.org/iptables/tree/include/xtables.h#n408).

Reply:
Thanks for the thorough explanation. Makes sense to me for this module thanks :)